build/changelog/ChangelogBuilder.php

```diff
@@ -0,0 +1,149 @@
+<?php
+namespace Aws\Build\Changelog;
+
+/**
+ * @internal
+ */
+
+class ChangelogBuilder 
+{
+  
+  public $newServiceFlag = False;
+
+  public function readChangelog($dir)
+  {
+    $dir = $dir."nextrelease/";
+    $changelogEntries = [];
+    if ( !is_dir($dir) || !$dh = opendir($dir) ) {
+        throw new \Exception("nextrelease directory doesn't exists or is not readable at location $dir");
+    } else {
+      //Ignore any files starting with a (.) dot 
+      $files=preg_grep('/^([^.])/', scandir($dir));
+      if ( empty($files) ) {
+        throw new \Exception("No release notes files found in $dir folder");
+      }
+      foreach ( $files as $file ) {
+        $str = file_get_contents($dir."/".$file);
+        $changelogEntries= array_merge($changelogEntries, $this->cleanJSON(json_decode($str)));
+      }
+      closedir($dh);
+    }
+    return $changelogEntries;
+  }
+
+  public function cleanJSON($arr) 
+  {
+    if ( empty($arr) || !is_array($arr) ) {
+      throw new \Exception("Invalid Input", 2);
+    } else {
+        $cleanedJSON=[];
+        foreach ( $arr as $x ) {
+          if ( $x->type == "NEW_SERVICE" ) {
```

Comment:
The NEW_SERVICE flag is set when a new service is being released  Besides that, all other updates except "DOC_UPDATE" are pushed to the changelog. 

Please, let me know why do u thing switch case would do better here.